tools/config-generator/main.go

```diff
@@ -301,9 +301,9 @@ func addMonitoringPubsubLabelsToJob(data *baseProwJobTemplateData, runID string)
 }
 
 // addVolumeToJob adds the given mount path as volume for the job.
-func addVolumeToJob(data *baseProwJobTemplateData, mountPath, name string, isSecret bool, content []string) {
+func addVolumeToJob(data *baseProwJobTemplateData, mountPath, name string, isSecret bool, isReadOnly bool, content []string) {
 	(*data).VolumeMounts = append((*data).VolumeMounts, []string{"- name: " + name, "  mountPath: " + mountPath}...)
-	if isSecret {
+	if isSecret && isReadOnly {
```

Comment:
Secrets are only readonly so should be an OR?

Reply:
well, secrets can be mounted as readwrite, as it is done in current proposal, so actually I guess here should be just check for readonly. Anyway, good point to update, thank you!

Reply:
Oh, I see now the intention was for this to only be for mount secrets as RW directories... it's confusing to understand in the other places. I think it'd be cleaner if instead of isSecret and isReadOnly variables, we had an enum:

```
Volume
SecretRO
SecretRW
```

or, I see little harm in all secrets being mounted as RW, which is even simpler! It's not like a job could harm another job via that method.

Reply:
Well, volumes can be mounted readwrite or readonly, and secret is just one possible source for volume. So for me the readonly and secret are just 2 different areas, I'd not mix them.

> I see little harm in all secrets being mounted as RW,

Best practices recommend not to do that, if possible :) However it will make the changes to the current setup smaller. So from my side, I'd still mount all current secrets readonly, as it is done now (except new ones). However of course it's up to maintainers to decide.

Reply:
I am deeply sorry, you were correct and I was wrong about mount of secrets and always readonly :) It's always readonly.
Only one (small) excuse for me that my test env bypasses this problem. Anyway, code is updated to resolve this challenge. Now I just copy k8s config to another directory, which is RW.

Reply:
Haha, no worries! I just know this config generator is already messy so I try to push back a bit on new features to it.